Do not use Files API to allow using on API 24+

Stypox opened the pull request:
Also removes external storage permission, now unneeded, since data is saved in app-specific files folder.
Unfortunately I haven't yet been able to test if everything works correctly: currently when I press the mic button nothing happens, but I guess this is due to the fact that deepspeech is downloading the model in the background without telling the user (correct me if I understood this wrong ;-) ).

Debug apk: [app-debug.zip](https://github.com/kamir86/MycroftDeepspeechCompanionApp/files/5612309/app-debug.zip)
Closes #1

Stypox committed:
Do not use Files API to allow using on API 24+

Also removes external storage permission, now unneeded, since data is saved in app-specific files folder

kamir86 commented:
> Also removes external storage permission, now unneeded, since data is saved in app-specific files folder.
> Unfortunately I haven't yet been able to test if everything works correctly: currently when I press the mic button nothing happens, but I guess this is due to the fact that deepspeech is downloading the model in the background without telling the user (correct me if I understood this wrong ;-) ).
> 
> Debug apk: [app-debug.zip](https://github.com/kamir86/MycroftDeepspeechCompanionApp/files/5612309/app-debug.zip)
> Closes #1

Seems good at glance. 
Currently no file is downloaded, but copied from assets to external dir, which is quite fast on my phone (1-2secs). Looks like that the deep speech c library is not fine to use internal files dir (so nothing happens due to prototipal error handling). 
I'll test this thanks

kamir86 commented:
Hello, this pr does not work cause this line https://github.com/kamir86/MycroftDeepspeechCompanionApp/blob/e4ea06ab05aca7d2986e396016631b762ca1cb2b/app/src/main/java/it/awt/mycroft/deepspeech/ui/main/SpeechFragment.kt#L111 asks for a removed permission, can you fix?

Stypox committed:
Do not ask for removed permissions

Stypox commented:
Fixed

Stypox commented:
I fixed the missing permission and removed the log calls I used for debugging (I realized they were cluttering the logcat).
I tested on my phone and STT works correctly :-D
